tests/test_unit/test_save_poses.py

```diff
@@ -219,33 +240,92 @@ def test_to_dlc_df_split_individuals(
-        request,
     ):
         """Test that the 'split_individuals' argument affects the behaviour
-        of the 'to_dlc_file` function as expected
+        of the 'to_dlc_file` function as expected.
         """
-
         with expected_exception:
             save_poses.to_dlc_file(
                 valid_pose_dataset,
-                new_dlc_h5_file,
+                new_h5_file,
                 split_individuals,
             )
-            ds = request.getfixturevalue("valid_pose_dataset")
-
+            # Get the names of the individuals in the dataset
+            ind_names = valid_pose_dataset.individuals.values
             # "auto" becomes False, default valid dataset is multi-individual
             if split_individuals in [False, "auto"]:
                 # this should save only one file
-                assert new_dlc_h5_file.is_file()
-                new_dlc_h5_file.unlink()
+                assert new_h5_file.is_file()
+                new_h5_file.unlink()
             elif split_individuals is True:
                 # this should save one file per individual
-                for ind in ds.individuals.values:
+                for ind in ind_names:
                     file_path_ind = Path(
-                        f"{new_dlc_h5_file.with_suffix('')}_{ind}.h5"
+                        f"{new_h5_file.with_suffix('')}_{ind}.h5"
                     )
                     assert file_path_ind.is_file()
                     file_path_ind.unlink()
+
+    def test_to_sleap_analysis_file_valid_dataset(
+        self, output_file_params, valid_pose_dataset, request
+    ):
+        """Test that saving a valid pose dataset to a valid/invalid
+        SLEAP-style file returns the appropriate errors."""
+        with output_file_params.get("to_sleap_file_expected_exception"):
+            file_fixture = output_file_params.get("file_fixture")
+            val = request.getfixturevalue(file_fixture)
+            file_path = val.get("file_path") if isinstance(val, dict) else val
+            save_poses.to_sleap_analysis_file(valid_pose_dataset, file_path)
+
+    def test_to_sleap_analysis_file_invalid_dataset(
+        self, invalid_pose_dataset, new_h5_file
+    ):
+        """Test that saving an invalid pose dataset to a valid
+        SLEAP-style file returns the appropriate errors."""
+        with pytest.raises(ValueError):
+            save_poses.to_sleap_analysis_file(
+                invalid_pose_dataset,
+                new_h5_file,
+            )
+
+    @pytest.mark.parametrize(
+        "sleap_h5_file, fps",
+        [
+            ("SLEAP_single-mouse_EPM.analysis.h5", 30),
+            ("SLEAP_three-mice_Aeon_proofread.analysis.h5", None),
+            ("SLEAP_three-mice_Aeon_mixed-labels.analysis.h5", 50),
+        ],
+    )
+    def test_to_sleap_analysis_file_returns_same_h5_file_content(
+        self, sleap_h5_file, fps, new_h5_file
+    ):
+        """Test that saving pose tracks from a SLEAP analysis file
```

Comment:
```suggestion
        """Test that saving pose tracks imported from a SLEAP analysis file
```

Reply:
Also, two more points about this function:
1. should it be an integration test?
2. will this work in conjunction with the other open PR? Imagine loading a single-animal analysis file with no assigned tracks. The changes introduced by the other PR, will "create" a new track, so when you save it the analysis file will not be the same. This is probably the desired behaviour, but it may break this test. FYI I haven't reviewed the other PR yet, so it could be that you have anticipated all of this and taken care of it. In which casem ignore my point.

Reply:
Now that I've seen the other PR, I know that this test won't break. You mock and test the "trackless" datasets separately